### PR TITLE
Conditionally show/hide UI features based on the tenant tier

### PR DESCRIPTION
**Pull request by pavinduLakshan**
### Purpose

$subject

### Related Issues
- N/A

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/extensions/components/subscription/api/subscription.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    RequestConfigInterface,
+    RequestErrorInterface,
+    RequestResultInterface 
+} from "apps/console/src/features/core/hooks/use-request";
+import { HttpMethods } from "modules/core/src/models";
+import { getDomainQueryParam } from "../../tenants/api/tenants";
+import { getTenantResourceEndpoints } from "../../tenants/configs";
+import { TenantTierRequestResponse } from "../models/subscription";
+
+export const useTenantTier = <Data = TenantTierRequestResponse,
```
Comment (brionmario) [inline apps/console/src/extensions/components/subscription/api/subscription.ts:29]: Lets add some Doc comments.
Refer the `oranizationDiscovery/api` feature.
```suggestion
export const useGetTenantTier = <Data = TenantTierRequestResponse, 
```

File: apps/console/src/extensions/components/subscription/contexts/subscription.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from "react";
+import { TenantTier } from "../models/subscription";
+
+export const SubscriptionContext: React.Context<{
```
Comment (brionmario) [inline apps/console/src/extensions/components/subscription/contexts/subscription.ts:22]: Import `Context` from react, so we can drop `React` import all together easily when we start using JSX runtime.

Also move the interface to be separate.

Ref: https://github.com/wso2/identity-apps/blob/master/apps/console/src/features/branding/context/branding-preference-context.tsx#L32
```suggestion
export const SubscriptionContext: React.Context<{
```

File: apps/console/src/extensions/components/subscription/api/subscription.ts
Comment (brionmario) [inline apps/console/src/extensions/components/subscription/api/subscription.ts:1]: File name should be the same as the default export.
Also use default exports in other places as well.
Check other files as well.

File: apps/console/src/extensions/components/subscription/api/subscription.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    const requestConfig: RequestConfigInterface = {
+        headers: {
+            Accept: "application/json",
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: getTenantResourceEndpoints().tenantSubscriptionApi + "/tier" + getDomainQueryParam()
+    };
+
+    const { data, error, isValidating, mutate } = useRequest<Data, Error>(requestConfig);
```
Comment (brionmario) [inline apps/console/src/extensions/components/subscription/api/subscription.ts:40]: Pass `shouldRetryOnError: false` to the config.

https://github.com/wso2/identity-apps/blob/master/apps/console/src/features/branding/api/use-get-custom-text-preference-fallbacks.ts#L66C9-L66C34

Lets see whether we can disable this globally later.

File: apps/console/src/extensions/components/subscription/contexts/subscription.ts
Comment (brionmario) [inline apps/console/src/extensions/components/subscription/contexts/subscription.ts:1]: Rename to `subscription-context.ts`

File: apps/console/src/extensions/components/subscription/providers/subscription-provider.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ */
+
+import React, { PropsWithChildren, ReactElement } from "react";
+import { SubscriptionContext } from "../contexts/subscription";
+import { TenantTier } from "../models/subscription";
+
+/**
+ * Subscription provider props interface.
+ */
+interface SubscriptionProviderPropsInterface {
```
Comment (brionmario) [inline apps/console/src/extensions/components/subscription/providers/subscription-provider.tsx:26]: export the interface.

Comment (pavinduLakshan) [reply apps/console/src/extensions/components/subscription/providers/subscription-provider.tsx:26]: Do we need to export this interface as it's used in only this file?

File: apps/console/src/extensions/components/subscription/api/subscription.ts
Comment (pavinduLakshan) [reply apps/console/src/extensions/components/subscription/api/subscription.ts:1]: In other similar places, files in the sub folders are named after the feature name though. [1][2][3]

[1] apps/console/src/extensions/components/api-resources/api/api-resources.ts
[2] apps/console/src/extensions/components/events/api/event-publishing.ts
[3] apps/console/src/extensions/components/feature-gate/api/feature-gate.ts

File: apps/console/src/extensions/components/subscription/api/subscription.ts (hunk 1 above)
Comment (pavinduLakshan) [reply apps/console/src/extensions/components/subscription/api/subscription.ts:29]: Any specific reason to use default exports? AFAIK, default exports are usually discouraged.

File: apps/console/src/extensions/components/subscription/contexts/subscription.ts
Comment (pavinduLakshan) [reply apps/console/src/extensions/components/subscription/contexts/subscription.ts:1]: Fixed with 2f9edfa

File: apps/console/src/extensions/components/subscription/api/subscription.ts (hunk 3 above)
Comment (pavinduLakshan) [reply apps/console/src/extensions/components/subscription/api/subscription.ts]: Fixed with 2f9edfa

File: apps/console/src/extensions/components/subscription/contexts/subscription.ts (hunk 2 above)
Comment (pavinduLakshan) [reply apps/console/src/extensions/components/subscription/contexts/subscription.ts]: Fixed with 2f9edfa

File: apps/console/src/extensions/components/subscription/providers/subscription-provider.tsx (hunk 4 above)
Comment (pavinduLakshan) [reply apps/console/src/extensions/components/subscription/providers/subscription-provider.tsx]: Fixed with 8caac5c

**Review by brionmario — APPROVED** [review]

**Pull request merged**